### PR TITLE
Add dev workflow support for azure compute module

### PR DESCRIPTION
**Pull request by metacpp**
This PR is to introduce the workflow for module developer:
1. test-kitchen/kitchen-terraform as test harness to run integration testing for terraform code.
2. Inspec/Rspec is used to write BDD style test code for module.
3. Rake task is introduced here as make/build system for module.
4. Integrate PR with TravisCI.

### CODE REVIEW

**Comment by metacpp** [comment]
@dcaro @echuvyrov can you take a look at the code review ?

**Comment by dtzar** [comment]
For reference - here is the Travis job output: https://travis-ci.org/Azure/terraform-azurerm-compute

@metacpp Your description still says "Don't merge it".  Is this still the case?  If not - can you please update?

**Comment by metacpp** [comment]
@dtzar after you approve it, i will merge it.

**Comment by metacpp** [comment]
@dtzar thanks for the suggestion. I agree with you since the "rake build" is triggered during PR, we need to make it as fast as possible. This is in my backlog and will update it later.

**Comment by metacpp** [comment]
The Linux build is pretty fast:
https://travis-ci.org/Azure/terraform-azurerm-compute/jobs/305971972

While the Mac OS X is much longer:
https://travis-ci.org/Azure/terraform-azurerm-compute/jobs/305971973

I don't whether there's docker image for os x... But we probably can speed up the build through removing the "brew update" every time.

**Comment by dtzar** [comment]
Can we just use the Linux build for now?  1 min is acceptable 👍

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# This file can be validated on:
+# http://lint.travis-ci.org/
+dist: trusty
+sudo: required
+language: ruby
+rvm:
+  - 2.3
+env:
+  global:
+  - TERRAFORM_VERSION=0.10.8
```
Comment (dtzar) [inline .travis.yml]: Minor thing - but want to update to 0.11.0 since this is the latest version?

Comment (metacpp) [reply .travis.yml]: the kitchen terraform only support v0.10.8.

**Comment by metacpp** [comment]
@dtzar I just disabled the os x from matrix. The brew install really slow, we probably need to figure out a better way before enable in PR validation.

File: .travis.yml (hunk 1 above)
Comment (metacpp) [reply .travis.yml]: I will check with them for the upgrade requirement.

**Pull request merged**